data-model-solr/src/main/java/org/mousephenotype/cda/solr/web/dto/AnatomyPageTableRow.java

```diff
@@ -48,59 +49,77 @@ public void  addIncrementToNumberOfImages(){
-        gene.setSymbol(image.getGeneSymbol());
-        gene.setAccessionId(image.getGeneAccession());
+        gene.setSymbol(observation.getGeneSymbol());
+        gene.setAccessionId(observation.getGeneAccession());
         MarkerBean allele = new MarkerBean();
-        allele.setSymbol(image.getAlleleSymbol());
-        allele.setSymbol(image.getAlleleAccession());
+        allele.setSymbol(observation.getAlleleSymbol());
+        allele.setSymbol(observation.getAlleleAccession());
         this.setGene(gene);
         this.setAllele(allele);
         this.setSexes(sex);
-        this.setDataSourceName(image.getDataSourceName());
-        this.setZygosity(image.getZygosity() != null ? ZygosityType.valueOf(image.getZygosity()) : ZygosityType.not_applicable);
+        this.setDataSourceName(observation.getDataSourceName());
+        this.setZygosity(observation.getZygosity() != null ? ZygosityType.valueOf(observation.getZygosity()) : ZygosityType.not_applicable);
         ImpressBaseDTO proc = new ImpressBaseDTO();
-        proc.setName(image.getProcedureName());
-        proc.setStableId(image.getProcedureStableId());
+        proc.setName(observation.getProcedureName());
+        proc.setStableId(observation.getProcedureStableId());
         ImpressBaseDTO param = new ImpressBaseDTO();
-        param.setName(image.getParameterName());
-        param.setStableId(image.getParameterStableId());
+        param.setName(observation.getParameterName());
+        param.setStableId(observation.getParameterStableId());
         this.setProcedure(proc);
         this.setParameter(param);
-        this.setPhenotypingCenter(image.getPhenotypingCenter());
+        this.setPhenotypingCenter(observation.getPhenotypingCenter());
 
+//        System.out.println("class="+observation.getClass());
         List<OntologyTerm> anatomyTerms = new ArrayList<>();
-        for (int i = 0; i < image.getAnatomyId().size(); i++){
-        	if (image.getExpression(image.getAnatomyId().get(i)).equalsIgnoreCase(expressionValue)){
-	        	OntologyTerm anatomy = new OntologyTerm();
-	        	anatomy.setId(new DatasourceEntityId(image.getAnatomyId().get(i), -1));
-	        	anatomy.setName(image.getAnatomyTerm().get(i));
-	        	anatomyTerms.add(anatomy);
-        	}
+        
+        if(!observation.getClass().getName().equals("org.mousephenotype.cda.solr.service.dto.ImageDTO")){
+       
+	        for (int i = 0; i < observation.getAnatomyId().size(); i++){
+	        	if (observation.getCategory().equalsIgnoreCase(expressionValue)){
+		        	OntologyTerm anatomy = new OntologyTerm();
+		        	anatomy.setId(new DatasourceEntityId(observation.getAnatomyId().get(i), -1));
+		        	anatomy.setName(observation.getAnatomyTerm().get(i));
+		        	anatomyTerms.add(anatomy);
+	        	}
+	        }
+        }
+        
+        if(observation.getClass().getName().equals("org.mousephenotype.cda.solr.service.dto.ImageDTO")){
```

Comment:
Might also consider some variation of:  if (observation instanceof ImageDTO) { ... }